src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -270,12 +310,43 @@ def get(self, request: Request, project: Project) -> Response:
                     )
                     and project.organization.flags.codecov_access
                 )
+
+                # Get commit sha from Git blame
+                commit_sha = (
+                    ctx["commit_id"]
+                    if ctx["commit_id"]
+                    else current_config["config"]["defaultBranch"]
+                )
+                try:
+                    should_get_commit_sha = should_get_codecov_data and (
+                        not ctx["commit_id"]
+                        or ctx["commit_id"] == current_config["config"]["defaultBranch"]
+                    )
+
+                    if should_get_commit_sha:
+                        integration = integrations.filter(provider="github")[0]
+                        line_no = ctx.get("line_no")
+                        if line_no:
+                            commit_sha = self.get_commit_sha(
+                                integration,
+                                project.organization_id,
+                                int(line_no),
+                                filepath,
+                                current_config["repository"],
+                                current_config["config"]["defaultBranch"],
+                            )
+                except Exception:
+                    logger.exception(
+                        "Failed to get commit sha from git blame, pending investigation. Continuing execution."
+                    )
+
+                # Get Codecov data
                 if should_get_codecov_data:
                     try:
                         result["lineCoverage"], result["codecovUrl"] = get_codecov_data(
                             repo=current_config["config"]["repoName"],
                             service=current_config["config"]["provider"]["key"],
-                            branch=current_config["config"]["defaultBranch"],
+                            branch=commit_sha,
```

Comment:
We will now use the commit sha (if it is available, else branch) to get codecov data (see [here](https://github.com/getsentry/sentry/pull/42978/files#diff-9ed2a929b59278cbb7d160ac361180e90dbb3ec7359f7543b614e12e1615d059R315-R319))